src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/corporate/AbstractCorporatePersonMapping.kt

```diff
@@ -0,0 +1,23 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.corporate
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.annotation.Transient
+import org.springframework.data.domain.Persistable
+import java.time.LocalDateTime
+
+abstract class AbstractCorporateMappingTyped<T>(
+  val label: String? = null,
+  val mappingType: CorporateMappingType,
+
+  @Transient
+  @Value("false")
+  val new: Boolean = true,
+
+  val whenCreated: LocalDateTime? = null,
+
+) : Persistable<T> {
+
+  override fun isNew(): Boolean = new
+}
+
+typealias AbstractCorporateMapping = AbstractCorporateMappingTyped<String>
```

Comment:
I guess you're expecting some where the type isn't String? If so would it be better to name this `CorporateMappingString`? If not then no need for generics?